internal/backends/winit/event_loop.rs

```diff
@@ -461,6 +465,14 @@ pub fn run() {
     // we have scheduled for a sync between the `Window` item properties the the winit window, so that we
     // apply them at `MainEventsCleared` time.
     let mut windows_with_pending_property_updates = Vec::new();
+    // With winit on Windows and with wasm, calling winit::Window::request_redraw() will not always deliver an
+    // Event::RedrawRequested (for example when the mouse cursor is outside of the window). So when we get woken
+    // up by the event loop to process new events from the operating system (NewEvents), we take note of all windows
+    // that called request_redraw() since the last iteration and we will call draw() ourselves, unless they received
+    // an Event::RedrawRequested in this new iteration. This vector collects the window ids of windows with pending
+    // redraw requests in the beginning of the loop iteration, removes ids that are covered by a windowing system
+    // supplied Event::RedrawRequested, and drains them for drawing at RedrawEventsCleared.
+    let mut windows_with_pending_redraw_requests = Vec::new();
```

Comment:
Why do we need that instead of always looping on ALL_WINDOWS from Event::RedrawEventsCleared
(i mean, we already loop over them, could we not also just check pending_redraw from there?)

Reply:
That was my first thought and attempt as well, but then I observed that it results in painting twice, which is bad performance wise.

This happens for example with the OpenGL underlay example:

1. The OS provides an `Event::RedrawRequested` event.
2. We draw and inside `draw()` `request_redraw()` gets called and `pending_redraw_request` becomes true.
3. Then `Events::RedrawEventsCleared` is processed and now we would see that `pending_redraw_request` is true, and we would call `draw()` again, painting twice.

That's why instead in `NewEvents` we collect, subtract for each OS supplied draw and draw the "remainder" (if there is any) in `RedrawEventsCleared`.